input-output-hk/lace
# feat(extension): use public key hash as id for opted in users

# Checklist

- [ ] JIRA - \<[LW-7574](https://input-output.atlassian.net/browse/LW-7574)>
- [ ] Fix Broken unit tests
- [ ] Add unit test for new methods added in user id service
---

## Proposed solution

This PR implements the hashed wallet public key as id for opted in users, to be able to keep tracking across browsers. For opted out users, the ID keeps being randomized

## Testing

- Start any of the setup flows and complete all the steps. 
- On the analytics step, agree to be tracked. 
- Go to post hog dashboard.
- For all steps before wallet creation you will see a random ID in post hog, after the wallet is created you will see an Alias event with a new ID. 
- Click on the ID and go to the events tab, you should see the previous event for the random ID, linked to this new ID

## Screenshots



[LW-7574]: https://input-output.atlassian.net/browse/LW-7574?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ


---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/1110/5648265212/index.html) for [b9005b86](https://github.com/input-output-hk/lace/pull/304/commits/b9005b867002ef682f67328fe7a32776b55c411a)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 4      | 0       | 0     | 35    | ❌     |
